### PR TITLE
Add harvester node driver to rancher

### PR DESCRIPTION
**Pull request by guangbochen**
**Problem:**
Will need to add the default harvester node driver to the rancher server.

**Solution:**
Add default harvester node driver to the rancher server.

**Related Issue:**
https://github.com/rancher/harvester/issues/513

**Test plan:**
- Check if the Harvester node driver exists in the Rancher and the UI is loaded successfully.

### CODE REVIEW

File: pkg/controller/master/rancher/harvester_node_driver.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	harvesterDriverName = "harvester"
+	driverURL           = "https://github.com/harvester/docker-machine-driver-harvester/releases/download/v0.1.0/docker-machine-driver-harvester-amd64.tar.gz"
+	driverUIURL         = "https://github.com/harvester/ui-driver-harvester/releases/download/v0.1.0/component.js"
+)
+
+func (h *Handler) AddHarvesterNodeDriver() (*rancherv3api.NodeDriver, error) {
```
Comment (gitlawr) [inline pkg/controller/master/rancher/harvester_node_driver.go:16]: Can we add upgrade logic for bumping the driver in the future?

Comment (guangbochen) [reply pkg/controller/master/rancher/harvester_node_driver.go:16]: done

File: pkg/controller/master/rancher/harvester_node_driver.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	driver, err := h.NodeDrivers.Get(harvesterDriverName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		logrus.Infof("Add default harvester node driver: %s", harvesterDriverName)
+		harvesterDriver := &rancherv3api.NodeDriver{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: harvesterDriverName,
+			},
+			Spec: rancherv3api.NodeDriverSpec{
+				DisplayName:      harvesterDriverName,
+				Description:      "build-in harvester node driver",
```
Comment (gitlawr) [inline pkg/controller/master/rancher/harvester_node_driver.go]: built-in

File: pkg/controller/master/rancher/harvester_node_driver.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		}
+
+		return h.NodeDrivers.Create(harvesterDriver)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if driver.Spec.URL == driverURL {
```
Comment (gitlawr) [inline pkg/controller/master/rancher/harvester_node_driver.go]: and compare driverUIURL

**Comment by guangbochen** [comment]
updated, thanks.

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**